source/MaterialXGenShader/ShaderGenerator.cpp

```diff
@@ -294,6 +294,50 @@ namespace
     }
 }
 
+void ShaderGenerator::registerShaderMetadata(const DocumentPtr& doc, GenContext& context) const
+{
+    ShaderMetadataRegistryPtr registry = context.getUserData<ShaderMetadataRegistry>(ShaderMetadataRegistry::USER_DATA_NAME);
```

Comment:
Do you need to clear the registry here to avoid accumulating if this is called more than once?

Reply:
In the call to `registry->addMetadata(...)`, inside the method there is a check that the same metadata has not been registered already.